typescript/sdk/src/deploy/middleware/TokenBridgeRouterDeployer.ts

```diff
@@ -0,0 +1,224 @@
+import { objMap, promiseObjAll } from '@hyperlane-xyz/sdk/src/utils/objects';
+import { utils } from '@hyperlane-xyz/utils';
+
+import { chainMetadata } from '../../consts/chainMetadata';
+import { HyperlaneCore } from '../../core/HyperlaneCore';
+import { MultiProvider } from '../../providers/MultiProvider';
+import {
+  TokenBridgeContracts,
+  TokenBridgeFactories,
+  tokenBridgeFactories,
+} from '../../tokenBridge';
+import { ChainMap, ChainName } from '../../types';
+import { HyperlaneRouterDeployer } from '../router/HyperlaneRouterDeployer';
+import { RouterConfig } from '../router/types';
+
+export enum BridgeAdapterType {
+  Mock = 'Mock',
+  Circle = 'Circle',
+}
+
+export interface CircleBridgeAdapterConfig {
+  type: BridgeAdapterType.Circle;
+  circleBridgeAddress: string;
+  messageTransmitterAddress: string;
+  usdcAddress: string;
+  circleDomainMapping: {
+    hyperlaneDomain: number;
+    circleDomain: number;
+  }[];
+}
+
+interface MockTokenBridgeAdapterConfig {
+  type: BridgeAdapterType.Mock;
+  mockTokenAddress: string;
+}
+
+export type BridgeAdapterConfig =
+  | CircleBridgeAdapterConfig
+  | MockTokenBridgeAdapterConfig;
+
+export type TokenBridgeConfig = RouterConfig & {
+  bridgeAdapterConfigs: BridgeAdapterConfig[];
+};
+
+export class TokenBridgeDeployer<
+  Chain extends ChainName,
+> extends HyperlaneRouterDeployer<
+  Chain,
+  TokenBridgeConfig,
+  TokenBridgeContracts,
+  TokenBridgeFactories
+> {
+  constructor(
+    multiProvider: MultiProvider<Chain>,
+    configMap: ChainMap<Chain, TokenBridgeConfig>,
+    protected core: HyperlaneCore<Chain>,
+    protected create2salt = 'tokenbridgedeployersalt',
+  ) {
+    super(multiProvider, configMap, tokenBridgeFactories, {});
+  }
+
+  async enrollRemoteRouters(
+    contractsMap: ChainMap<Chain, TokenBridgeContracts>,
+  ): Promise<void> {
+    // Enroll the TokenBridgeRouter with each other
+    await super.enrollRemoteRouters(contractsMap);
+
+    // Enroll the circle adapters with each other
+    const deployedChains = Object.keys(contractsMap);
+    await promiseObjAll(
+      objMap(contractsMap, async (local, contracts) => {
+        const chainConnection = this.multiProvider.getChainConnection(local);
+        // only enroll chains which are deployed
+        const enrollChains = this.multiProvider
+          .remoteChains(local)
+          .filter((c) => deployedChains.includes(c));
+        for (const remote of enrollChains) {
+          const remoteDomain = chainMetadata[remote].id;
+          if (
+            !contracts.circleBridgeAdapter ||
+            !contractsMap[remote].circleBridgeAdapter
+          ) {
+            continue;
+          }
+          const current = await contracts.circleBridgeAdapter.routers(
+            remoteDomain,
+          );
+          const expected = utils.addressToBytes32(
+            contractsMap[remote].circleBridgeAdapter!.address,
+          );
+          if (current !== expected) {
```

Comment:
Code's going far to the right here, consider inverting this if (`if === then continue`)